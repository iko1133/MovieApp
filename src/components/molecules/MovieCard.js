import React from "react";
import styled from "styled-components/native";

import * as CONSTANTS from "config/constants";
import Eye from "assets/icons/Eye";
import FilledHeart from "assets/icons/FilledHeart";

const Container = styled.TouchableOpacity`
  flex-direction: row;
`;

const Image = styled.Image`
  aspect-ratio: 1;
  border-radius: 6px;
  width: 100%;
`;

const FavoriteButton = styled.TouchableOpacity`
  align-items: center;
  background-color: ${({ theme }) => theme.MOVIE_CARD_INFO_BACKGROUND_COLOR};
  border-radius: 8px;
  height: 40px;
  justify-content: center;
  top: 4px;
  right: 4px;
  position: absolute;
  width: 40px;
`;

const HideMovieButton = styled.TouchableOpacity`
  align-items: center;
  background-color: ${({ theme }) => theme.MOVIE_CARD_INFO_BACKGROUND_COLOR};
  border-radius: 8px;
  height: 40px;
  justify-content: center;
  top: 54px;
  right: 4px;
  padding: 8px;
  position: absolute;
  width: 40px;
`;

const NameContainer = styled.View`
  align-items: center;
  bottom: 0px;
  background-color: ${({ theme }) => theme.MOVIE_CARD_INFO_BACKGROUND_COLOR};
  padding: 4px 8px;
  position: absolute;
  width: 100%;
`;

const MovieTitle = styled.Text`
  color: ${({ theme }) => theme.MOVIE_CARD_INFO_TEXT_COLOR};
  font-size: 18px;
`;

/**
 * A card displaying movie object poster and title, while having several on press options
 * @param {object} style - Styling of the card itself
 * @param {function} onPress - A callback after the card is pressed
 * @param {object} item - Object describing the given item
 * @param {boolean} isFavorite - A boolean, indicating whether item displayed is favorite or not
 * @param {function} onHeartPress - A callback after heart icon is pressed
 * @param {function} onHidePress - A callback after eye icon is pressed
 */
const MovieCard = ({
  style,
  onPress,
  item,
  isFavorite,
  onHeartPress,
  onHidePress,
}) => {
  return (
    <Container style={style} onPress={onPress} activeOpacity={0.9}>
      <Image
        source={{
          uri: item.image ? item.image : CONSTANTS.IMAGE_NOT_FOUND_IMAGE,
        }}
        // resizeMode="contain"
      />
      <FavoriteButton onPress={onHeartPress}>
        <FilledHeart filled={isFavorite} />
      </FavoriteButton>

      <HideMovieButton onPress={onHidePress}>
        <Eye />
      </HideMovieButton>

      <NameContainer>
        <MovieTitle>{item.title}</MovieTitle>
      </NameContainer>
    </Container>
  );
};

export default MovieCard;
