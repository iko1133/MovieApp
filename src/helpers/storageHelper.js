import AsyncStorage from "@react-native-async-storage/async-storage";

/**
 * A function that will save given object under given key
 * @param {object} object - An object that needs to be saved in storage
 * @param {string} key - A string key the object should be saved under
 * @param {string} errorText - A string that will be displayed if something goes wrong
 */
export const storeObject = async (
  object,
  key,
  errorText = "There was an error while trying to save"
) => {
  try {
    const jsonObj = JSON.stringify(object);
    await AsyncStorage.setItem(key, jsonObj);
  } catch (e) {
    alert(errorText);
  }
};

/**
 * A function that will return an object saved under given key or null if none was found
 * @param {string} key - A string key that desired object should be saved under
 * @param {string} errorText - A string that will be displayed if something goes wrong
 * @returns Object under given key or null if none was found
 */
export const getObject = async (key, errorText) => {
  try {
    const jsonVal = await AsyncStorage.getItem(key);
    return jsonVal != null ? JSON.parse(jsonVal) : null;
  } catch (e) {
    alert(errorText);
  }
};
