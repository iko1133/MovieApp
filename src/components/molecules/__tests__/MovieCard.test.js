import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import MovieCard from "../MovieCard";

const items = [
  {
    id: "tt0411008",
    resultType: "Title",
    image:
      "https://imdb-api.com/images/original/MV5BNzhlY2E5NDUtYjJjYy00ODg3LWFkZWQtYTVmMzU4ZWZmOWJkXkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_Ratio0.7273_AL_.jpg",
    title: "Lost",
    description: "(2004) (TV Series)",
  },
  {
    id: "tt0110413",
    resultType: "Title",
    image:
      "https://imdb-api.com/images/original/MV5BODllNWE0MmEtYjUwZi00ZjY3LThmNmQtZjZlMjI2YTZjYmQ0XkEyXkFqcGdeQXVyNTc1NTQxODI@._V1_Ratio0.7273_AL_.jpg",
    title: "LÃ©on: The Professional",
    description: "(1994)",
  },
];

describe("Testing MovieCard", () => {
  it("Renders MovieCard", () =>
    expect(
      renderer.create(WrapInThemeProvider(<MovieCard item={items[0]} />))
    ).toMatchSnapshot());

  it("Renders movie card with given item", () =>
    expect(
      renderer.create(WrapInThemeProvider(<MovieCard item={items[1]} />))
    ).toMatchSnapshot());
});
