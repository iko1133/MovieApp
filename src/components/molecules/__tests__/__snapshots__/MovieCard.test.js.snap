// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieCard Renders MovieCard 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://imdb-api.com/images/original/MV5BNzhlY2E5NDUtYjJjYy00ODg3LWFkZWQtYTVmMzU4ZWZmOWJkXkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_Ratio0.7273_AL_.jpg",
      }
    }
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "borderRadius": 6,
          "width": "100%",
        },
      ]
    }
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F1F1FAA",
        "borderRadius": 8,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 4,
        "top": 4,
        "width": 40,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={24}
      bbWidth={24}
      focusable={false}
      height={24}
      stroke="#FFF"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        propList={
          Array [
            "stroke",
          ]
        }
        stroke={4294967295}
      >
        <RNSVGPath
          d="M12 4.248C8.852-1.154 0 .423 0 7.192 0 11.853 5.571 16.619 12 23c6.43-6.381 12-11.147 12-15.808C24 .4 15.125-1.114 12 4.248z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F1F1FAA",
        "borderRadius": 8,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "position": "absolute",
        "right": 4,
        "top": 54,
        "width": 40,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={20}
      bbWidth={20}
      focusable={false}
      height={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup>
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4294967295}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={1.5}
        >
          <RNSVGPath
            d="m12.109 7.892-4.217 4.217a2.982 2.982 0 1 1 4.217-4.217Z"
          />
          <RNSVGPath
            d="M14.846 4.808A8.026 8.026 0 0 0 10 3.108c-2.941 0-5.683 1.733-7.592 4.733a4.438 4.438 0 0 0 0 4.325 11.939 11.939 0 0 0 2.259 2.642M7.017 16.275a7.679 7.679 0 0 0 2.983.617c2.942 0 5.683-1.733 7.592-4.733a4.438 4.438 0 0 0 0-4.325 13.518 13.518 0 0 0-.883-1.226"
          />
          <RNSVGPath
            d="M12.925 10.583a2.971 2.971 0 0 1-2.35 2.35M7.892 12.108l-6.225 6.225M18.333 1.667l-6.225 6.225"
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1F1F1FAA",
          "bottom": 0,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          },
        ]
      }
    >
      Lost
    </Text>
  </View>
</View>
`;

exports[`Testing MovieCard Renders movie card with given item 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://imdb-api.com/images/original/MV5BODllNWE0MmEtYjUwZi00ZjY3LThmNmQtZjZlMjI2YTZjYmQ0XkEyXkFqcGdeQXVyNTc1NTQxODI@._V1_Ratio0.7273_AL_.jpg",
      }
    }
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "borderRadius": 6,
          "width": "100%",
        },
      ]
    }
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F1F1FAA",
        "borderRadius": 8,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 4,
        "top": 4,
        "width": 40,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={24}
      bbWidth={24}
      focusable={false}
      height={24}
      stroke="#FFF"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        propList={
          Array [
            "stroke",
          ]
        }
        stroke={4294967295}
      >
        <RNSVGPath
          d="M12 4.248C8.852-1.154 0 .423 0 7.192 0 11.853 5.571 16.619 12 23c6.43-6.381 12-11.147 12-15.808C24 .4 15.125-1.114 12 4.248z"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F1F1FAA",
        "borderRadius": 8,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "position": "absolute",
        "right": 4,
        "top": 54,
        "width": 40,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={20}
      bbWidth={20}
      focusable={false}
      height={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup>
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4294967295}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={1.5}
        >
          <RNSVGPath
            d="m12.109 7.892-4.217 4.217a2.982 2.982 0 1 1 4.217-4.217Z"
          />
          <RNSVGPath
            d="M14.846 4.808A8.026 8.026 0 0 0 10 3.108c-2.941 0-5.683 1.733-7.592 4.733a4.438 4.438 0 0 0 0 4.325 11.939 11.939 0 0 0 2.259 2.642M7.017 16.275a7.679 7.679 0 0 0 2.983.617c2.942 0 5.683-1.733 7.592-4.733a4.438 4.438 0 0 0 0-4.325 13.518 13.518 0 0 0-.883-1.226"
          />
          <RNSVGPath
            d="M12.925 10.583a2.971 2.971 0 0 1-2.35 2.35M7.892 12.108l-6.225 6.225M18.333 1.667l-6.225 6.225"
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1F1F1FAA",
          "bottom": 0,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          },
        ]
      }
    >
      LÃ©on: The Professional
    </Text>
  </View>
</View>
`;
